src/cargo/ops/cargo_compile.rs

```diff
@@ -246,16 +247,27 @@ fn scrape_target_config(config: &Config, triple: &str)
         let table = try!(config.get_table(&key)).unwrap().0;
         for (k, _) in table.into_iter() {
             let key = format!("{}.{}", key, k);
-            let (v, path) = try!(config.get_string(&key)).unwrap();
-            if k == "rustc-flags" {
-                let whence = format!("in `{}` (in {:?})", key, path);
-                let (paths, links) = try!(
-                    BuildOutput::parse_rustc_flags(&v, &whence)
-                );
-                output.library_paths.extend(paths.into_iter());
-                output.library_links.extend(links.into_iter());
-            } else {
-                output.metadata.push((k, v));
+            match try!(config.get(&key)).unwrap() {
+                ConfigValue::String(v, path) => {
+                    if k == "rustc-flags" {
+                        let whence = format!("in `{}` (in {:?})", key, path);
+                        let (paths, links) = try!(
+                            BuildOutput::parse_rustc_flags(&v, &whence)
+                        );
+                        output.library_paths.extend(paths.into_iter());
+                        output.library_links.extend(links.into_iter());
+                    } else {
+                        output.metadata.push((k, v));
+                    }
+                },
+                ConfigValue::List(a, _) => {
+                    if k == "rustc-link-lib" {
+                        output.library_links.extend(a.into_iter().map(|(v, _)| v));
+                    } else if k == "rustc-link-search" {
+                        output.library_paths.extend(a.into_iter().map(|(v, _)| PathBuf::new(&v)));
+                    }
+                },
+                _ => { }
```

Comment:
The semantics here have changed a little subtly, we may want to continue to emit errors for other types of configuration and also continue to emit errors for arrays of configuration in `rustc-link-lib` and `rustc-link-search`.